a10_octavia/tests/unit/controller/worker/tasks/test_service_group_tasks.py

```diff
@@ -36,9 +41,32 @@ def setUp(self):
         super(TestHandlerServiceGroupTasks, self).setUp()
         imp.reload(task)
         self.client_mock = mock.Mock()
+        self.conf = self.useFixture(oslo_fixture.Config(cfg.CONF))
+        self.conf.register_opts(A10_SERVICE_GROUP_OPTS,
+                                group=a10constants.SERVICE_GROUP_CONF_SECTION)
+
+    def tearDown(self):
+        super(TestHandlerServiceGroupTasks, self).tearDown()
+        self.conf.reset()
 
     def test_revert_pool_create_task(self):
         mock_pool = task.PoolCreate()
         mock_pool.axapi_client = self.client_mock
         mock_pool.revert(POOL, VTHUNDER)
         self.client_mock.slb.service_group.delete.assert_called_with(POOL.id)
+
+    def test_create_lb_algorithm_source_ip_hash_only(self):
+        mock_pool = task.PoolCreate()
+        mock_pool.axapi_client = self.client_mock
+        mock_pool.CONF = self.conf
+        POOL = o_data_models.Pool(id=a10constants.MOCK_POOL_ID,
```

Comment:
Just a minor comment. Can't we just do an update of global POOL object for keys protocol and lb_algorithm instead of instantiating a new POOL object ?

Reply:
Yes, its question actually, does that affects entire POOL object OR its limited to that function only.

Reply:
As per discussion, we will not create new instance of object if we are not making change in it, So, I am marking this resolved.
We can follow similar format in future as well.